### PR TITLE
Update Persistence and change storage shape

### PR DESCRIPTION
**Pull request by scme0**

### CODE REVIEW

**Review by kevjt — APPROVED** [review]
nice change

File: source/Octopus.Tentacle/Kubernetes/Diagnostics/KubernetesAgentMetrics.cs [hunk 1]
```diff
@@ -1,80 +1,80 @@
...
+                await PerformUpdateOnMetricsForReason(reason, reasonDictionary =>
                 {
-                    var sourceEventsForReason = LoadFromPersistence(reason) ?? new Dictionary<string, List<DateTimeOffset>>();
-
-                    if (!sourceEventsForReason.TryGetValue(source, out var occurenceTimestamps))
+                    if (!reasonDictionary.TryGetValue(source, out var occurenceTimestamps))
                     {
-                        occurenceTimestamps = new List<DateTimeOffset>();
-                        sourceEventsForReason[source] = occurenceTimestamps;
+                        reasonDictionary[source] = occurenceTimestamps = new List<DateTimeOffset>();
```
Comment (rain-on) [inline source/Octopus.Tentacle/Kubernetes/Diagnostics/KubernetesAgentMetrics.cs:39]: I cannot quite reason about this line - why are there 2 assignments?
Assume it makes all 3 the same thing i.e. a new list?

Comment (rain-on) [reply source/Octopus.Tentacle/Kubernetes/Diagnostics/KubernetesAgentMetrics.cs:39]: Accepted in outside discussions

**Review by rain-on — APPROVED** [review]
LGTM

**Pull request merged**